src/features/courses/course-repository.ts

```diff
@@ -263,6 +263,36 @@ class CourseRepository {
         }
     }
 
+    async extendTopicByUser(options: ExtendTopicForUserOptions): Promise<UpsertResult<StudentTopicOverride>> {
+        try {
+            const found = await StudentTopicOverride.findOne({where: {...options.where, active: true}});
+            if (!found) {
+                const newExtension = await StudentTopicOverride.create({...options.where, ...options.updates}, {validate: true});
+
+                return {
+                    createdNewEntry: true,
+                    updatedCount: 1,
+                    updatedRecords: [newExtension],
+                };
+            }
+            const updates = await StudentTopicOverride.update(options.updates, {
+                where: {
+                    ...options.where,
+                    active: true
+                },
+                returning: true,
+            });
+            return {
+                createdNewEntry: false,
+                updatedCount: updates[0],
+                updatedRecords: updates[1],
+            };
+        } catch (e) {
+            this.checkCourseTopicError(e);
+            throw new WrappedError(Constants.ErrorMessage.UNKNOWN_APPLICATION_ERROR_MESSAGE, e);
```

Comment:
Might be more clear to make this message be something like:
`Could not extend topic`
That message is for us and we could use it at a glance to tell what feature is broken in the logs
I tend to keep `UNKNOWN_APPLICATION_ERROR_MESSAGE` for issues that should never happen, i.e. nulls/undefineds that have already been check but with strict it has to be checked in the callback again

Reply:
Should we have a constants file for error messages?

Reply:
(Besides the seemingly generic ones in Constants.ErrorMessage)

Reply:
We could if it grew larger but I don't think we need it
In this case, it's an internal server error so this message is just for us
Additionally, this error message shouldn't be reused